lib/builder.js

```diff
@@ -143,10 +143,15 @@ module.exports = class Builder extends EventEmitter {
         }),
       () => {
         this._cancelationRequest = null;
+        this._settleNodes();
       }
     );
   }
 
+  _settleNodes() {
+    this.nodeWrappers.forEach(nw => nw.settle());
```

Comment:
If watchman triggers `revise` after the specific node is invoked, but before the build completes, this approach will loose that information, resulting in erroneous skipped plugin build steps.

The other PR does not require 2 phase marking or cleanup, being less error prone is an important detail.